model/preference.go

```diff
@@ -102,10 +102,8 @@ func (o *Preference) PreUpdate() {
 	if o.Category == PreferenceCategoryTheme {
 		// decode the value of theme (a map of strings to string) and eliminate any invalid values
 		var props map[string]string
-		if err := json.NewDecoder(strings.NewReader(o.Value)).Decode(&props); err != nil {
-			// just continue, the invalid preference value should get caught by IsValid before saving
-			return
-		}
+		// just continue, the invalid preference value should get caught by IsValid before saving
+		json.NewDecoder(strings.NewReader(o.Value)).Decode(&props)
 
 		colorPattern := regexp.MustCompile(`^#[0-9a-fA-F]{3}([0-9a-fA-F]{3})?$`)
```

Comment:
This call to `regexp.MustCompile` everytime is bad for performance. Can we move it outside and make it a global var? (Slightly out of scope but requires very minimal changes)